access/validator.go

```diff
@@ -227,9 +228,16 @@ func (v *TransactionValidator) checkExpiry(tx *flow.TransactionBody) error {
 	return nil
 }
 
+type noopMemoryGauge struct {
+}
+
+func (noopMemoryGauge) MeterMemory(_ common.MemoryUsage) error {
+	return nil
+}
+
 func (v *TransactionValidator) checkCanBeParsed(tx *flow.TransactionBody) error {
 	if v.options.CheckScriptsParse {
-		_, err := parser2.ParseProgram(string(tx.Script))
+		_, err := parser2.ParseProgram(string(tx.Script), noopMemoryGauge{})
```

Comment:
It should be possible to pass nil here. If not, we should fix this on the Cadence side.

If it is needed at the moment, maybe declare a global for the noop gauge and re-use it across calls.